src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.SqlClient.cs

```diff
@@ -816,5 +816,144 @@ static internal ArgumentOutOfRangeException InvalidDataRowVersion(DataRowVersion
+            return e;
+        }
+
+        internal static ArgumentException InvalidPrefixSuffix()
+        {
+            ArgumentException e = new ArgumentException(SR.GetString(SR.ADP_InvalidPrefixSuffix));
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+
+        // the return value is true if the string was quoted and false if it was not
+        // this allows the caller to determine if it is an error or not for the quotedString to not be quoted
+        internal static bool RemoveStringQuotes(string quotePrefix, string quoteSuffix, string quotedString, out string unquotedString)
+        {
+            int prefixLength;
+            if (quotePrefix == null)
+            {
+                prefixLength = 0;
+            }
+            else
+            {
+                prefixLength = quotePrefix.Length;
+            }
+
+            int suffixLength;
+            if (quoteSuffix == null)
+            {
+                suffixLength = 0;
+            }
+            else
+            {
+                suffixLength = quoteSuffix.Length;
+            }
+
+            if ((suffixLength + prefixLength) == 0)
+            {
+                unquotedString = quotedString;
+                return true;
+            }
+
+            if (quotedString == null)
+            {
+                unquotedString = quotedString;
+                return false;
+            }
+
+            int quotedStringLength = quotedString.Length;
+
+            // is the source string too short to be quoted
+            if (quotedStringLength < prefixLength + suffixLength)
+            {
+                unquotedString = quotedString;
+                return false;
+            }
+
+            // is the prefix present?
+            if (prefixLength > 0)
+            {
+                if (!quotedString.StartsWith(quotePrefix, StringComparison.Ordinal))
+                {
+                    unquotedString = quotedString;
+                    return false;
+                }
+            }
+
+            // is the suffix present?
+            if (suffixLength > 0)
+            {
+                if (!quotedString.EndsWith(quoteSuffix, StringComparison.Ordinal))
+                {
+                    unquotedString = quotedString;
+                    return false;
+                }
+                unquotedString = quotedString.Substring(prefixLength, quotedStringLength - (prefixLength + suffixLength)).Replace(quoteSuffix + quoteSuffix, quoteSuffix);
+            }
+            else
+            {
+                unquotedString = quotedString.Substring(prefixLength, quotedStringLength - prefixLength);
+            }
+            return true;
+        }
+
+        internal static ArgumentOutOfRangeException InvalidCommandBehavior(CommandBehavior value)
+        {
+#if DEBUG
+            if ((0 <= (int)value) && ((int)value <= 0x3F))
+            {
+                Debug.Assert(false, "valid CommandType " + value.ToString());
```

Comment:
Nit: The if conditions can be removed and this can be `Debug.Assert(!((0 <= (int)value) && ((int)value <= 0x3F)), "valid CommandType " + value.ToString());`